scripts/operatorhub-deploy.sh

```diff
@@ -0,0 +1,173 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail
+
+# Colors for better user experience
+WHITE='\033[1;37m'
+NC='\033[0m' # No color
+
+# This script helps in deployment of new version of Eunomia to OperatorHub
+usage() {
+    cat <<EOT
+Usage: operatorhub-deploy.sh [--send-pr] VERSION
+
+Creates a fork of https://github.com/operator-framework/community-operators
+and pushes changes with a new Eunomia version to it. Then a PR can be
+manually created.
+
+VERSION in format x.y.z, e.g. 0.1.2
+
+With --send-pr flag set, a Pull Request with a new Eunomia version in
+https://github.com/operator-framework/community-operators will be
+created.
+
+The script will modify the contents of the deploy/ directory - a
+directory structure deploy/olm-catalog/eunomia will be created. Inside
+files necessary to deploy Eunomia to OperatorHub will be created. When
+the script completes, the generated files get cleaned up.
+EOT
+}
+
+if [[ "${1:-}" == --send-pr ]]; then
+    send_pr=true
+    shift
+elif [[ "${1:-}" =~ ^-.*$ ]]; then
+    cat <<EOT
+
+error: You issued operatorhub-deploy.sh $@
+Did you want to set the --send-pr flag?
+
+EOT
+    usage
+    exit 1
+fi
+
+if ! [[ "${1:-}" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+    echo -e "\nerror: VERSION argument not passed, or in wrong format\n" >&2
+    usage
+    exit 1
+fi
+new_version="$1"
+eunomia_root=$(
+    cd "$(dirname "$0")/.."
+    pwd
+)
+
+# "operator-sdk olm-catalog gen-csv" command requires a specific project layout:
+# https://github.com/operator-framework/operator-sdk/blob/v0.8.1/doc/user/olm-catalog/generating-a-csv.md#configuration
+echo -e "${WHITE}Creating directory structure to satisfy operator-sdk"
+echo -e "\n- Generating operator.yaml and role.yaml${NC}"
+# Create temporary directory to store intermediate helm-generated files
+tmp_dir=$(mktemp -d)
+helm template "${eunomia_root}/deploy/helm/eunomia-operator/" \
+    --set eunomia.operator.deployment.operatorHub=true \
+    --set eunomia.operator.image.tag="v${new_version}" \
+    --output-dir "$tmp_dir"
+# Create operator.yaml and role.yaml
+cat "${tmp_dir}/eunomia-operator/templates/deployment.yaml" \
+    >"${eunomia_root}/deploy/operator.yaml"
+cat "${tmp_dir}/eunomia-operator/templates/clusterrole-operators.yaml" \
+    "${tmp_dir}/eunomia-operator/templates/role-operator.yaml" \
+    >"${eunomia_root}/deploy/role.yaml"
+# Delete temporary directory
+rm -rf "$tmp_dir"
+
+# Get previous versions from OperatorHub
+echo -e "\n${WHITE}- Downloading previous releases of Eunomia from OperatorHub${NC}"
+tmp_dir=$(mktemp -d)
+
+git -C "$tmp_dir" clone https://github.com/operator-framework/community-operators
+git_context="git -C ${tmp_dir}/community-operators"
+hub_context="hub -C ${tmp_dir}/community-operators"
+
+mkdir -p "${eunomia_root}/deploy/olm-catalog"
```

Comment:
[optional] how about preceding this line with a check verifying that the directory doesn't exist? if it's there when the script starts, any files in it would get carried over to the final PR, IIUC ?